src/sentry/services/http.py

```diff
@@ -140,6 +143,16 @@ def prepare_environment(self, env=None):
         for k, v in convert_options_to_env(self.options):
             env.setdefault(k, v)
 
+        # If there are multiple UWSGI_{protocol}_SOCKET options set,
+        # we want to remove any other than the ones we care about. We
+        # explicitly don't want to confusingly bind multiple sockets with
+        # different protocols. This is mostly relevant for devserver
+        # when forking for the subprocesses that inherit the parent's env
+        wanted_protocol = self.options["protocol"]
+        for protocol in self.protocol_options:
+            if protocol != wanted_protocol:
+                env.pop("UWSGI_%s_SOCKET" % protocol.upper(), None)
```

Comment:
Wouldn't this be more comprehensive (also more expressive for me):
```suggestion
        wanted_socket = "UWSGI_%s_SOCKET" % self.options["protocol"].upper()
        unwanted_sockets = set(key for key in env.keys() if key.startswith('UWSGI_') and key.endswith('_SOCKET'))
        unwanted_sockets.remove(wanted_socket)
        for key in unwanted_sockets:
            del env[key]
```

Reply:
It would catch potentially _other_ protocols, but I'm not really trying to prevent anyone from shooting themselves in the foot here by going out of their way to do other stuff.

Reply:
> but I'm not really trying to prevent anyone from shooting themselves in the foot here by going out of their way to do other stuff.

I guess this is fair.